### PR TITLE
Technical docs SEO guide

### PR DESCRIPTION
**Pull request by davidfischer**
* This PR adds a guide about SEO for technical docs with a focus on Sphinx. It includes the basics of SEO and how RTD and Sphinx help with that.
* A lot of this is explanations with links to existing other docs like the sitemaps guide, canonical guide, redirects, GA guide, etc. I think that's a good thing!
* One section that I would like to add but haven't yet is a section on how good technical writing should naturally lead to better SEO. That could almost be its own guide, but I would like to include stuff like:
  * Using words and phrases that users use (and would search for) without keyword stuffing. This will help users search for stuff.
  * Writing with simpler language will also help search engines understand content.
  * Having different types of docs (how-to guides, tutorials, reference docs) and how that will cover different things people search for

### CODE REVIEW

File: docs/guides/technical-docs-seo-guide.rst [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+relative to most of the internet.
+
+
+Optimizing your docs for search engine spiders
+----------------------------------------------
+
+Once a crawler or spider finds your site, it will follow links and redirects
+in an attempt to find any and all pages on your site.
+While there are a few ways to guide the search engine in its crawl
+for example by using a sitemap or a robots.txt file which we'll discuss shortly,
```
Comment (humitos) [inline docs/guides/technical-docs-seo-guide.rst]: Not sure it worth, but we may want to link `sitemap` and `robots.txt` to each section.

File: docs/guides/technical-docs-seo-guide.rst [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+the most important thing is making sure the spider can follow links on your site
+and get to all your pages.
+
+Avoid orphan pages
+~~~~~~~~~~~~~~~~~~
+
+Sphinx calls pages that don't have links to them "orphans"
+and will throw a warning while building documentation that contains an orphan
+unless the warning is silenced with the :ref:`orphan directive <sphinx:metadata>`:
```
Comment (humitos) [inline docs/guides/technical-docs-seo-guide.rst]: For some reason, this `:ref:` is rendering as https://sphinx.readthedocs.io/en/latest/usage/restructuredtext/field-lists.html#metadata instead of https://sphinx-doc.org

File: docs/guides/technical-docs-seo-guide.rst [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+::
```
Comment (humitos) [inline docs/guides/technical-docs-seo-guide.rst]:
```suggestion
.. prompt:: bash
```

Should disable selecting the `$` and render the output accordingly.

File: docs/guides/technical-docs-seo-guide.rst [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+For example, if you do have a video in your docs,
+make sure the rest of that page describes the content of the video.
+
+When using images, make sure to set the image alt text or set a caption on figures.
+For Sphinx, the image and figure directives support this:
+
+.. sourcecode:: rst
+
+    .. image:: your-image.png
+        :align: right
```
Comment (humitos) [inline docs/guides/technical-docs-seo-guide.rst]: I think we can remove the `align` option from here so it's cleaner, since that option is not related with what we are trying to show in the example.

File: docs/guides/technical-docs-seo-guide.rst [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+This guide isn't meant to be your only resource on SEO,
+and there's a lot of SEO topics not covered here.
+For additional reading, please see the
+:ref:`external resources <guides/technical-docs-seo-guide:External resources>` section.
+
+While many of the topics here apply to all forms of technical documentation,
+this guide will focus on Sphinx, which is the most common
+documentation authoring tool on Read the Docs,
+as well as improvements provided by Read the Docs itself.
```
Comment (humitos) [inline docs/guides/technical-docs-seo-guide.rst]: Considering the document has a few sections and subsections, it may be good to have a TOC after this paragraph.

```suggestion
as well as improvements provided by Read the Docs itself.


.. contents:: Table of contents
   :local:
   :backlinks: none
   :depth: 3
```

File: docs/guides/technical-docs-seo-guide.rst [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+Read the Docs supports :doc:`setting the canonical URL </guides/canonical>`
+if you are using a :doc:`custom domain </custom_domains>`
+under :guilabel:`Admin` > :guilabel:`Domains`
+in the Read the Docs dashboard.
+
+
+Use a robots.txt file
+~~~~~~~~~~~~~~~~~~~~~
+
+A ``robots.txt`` file is readable by crawlers
```
Comment (humitos) [inline docs/guides/technical-docs-seo-guide.rst]: We don't have a full Guide explaining how `robots.txt` works, but we could link this FAQ somewhere in this section: https://docs.readthedocs.io/en/latest/faq.html#how-can-i-avoid-search-results-having-a-deprecated-version-of-my-docs

File: docs/guides/technical-docs-seo-guide.rst [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+and lives at the root of your site (eg. https://docs.readthedocs.io/robots.txt).
+It tells search engines which pages to crawl or not to crawl
+and can allow you to control how a search engine crawls your site.
+For example, you may want to request that search engines not index
+unsupported versions of your documentation while keeping those docs online in case people need them.
+
+By default, Read the Docs serves a ``robots.txt`` for you
+which points to a :ref:`sitemap <guides/technical-docs-seo-guide:Use a sitemap.xml file>`
```
Comment (humitos) [inline docs/guides/technical-docs-seo-guide.rst]: I think I'd remove the mention to sitemap here, or I'd introduce the concept first.

When reading the guide, I felt confusing the mention to sitemap at this point without mentioning it previously or explaining anything about it.

Removing the first sentence al leaving only "To customize this file,..." works to me.

Also, the "By default," sentence can be merged into the following section (Use a sitemap.xml) where robots.txt was already presented and now we can mentioned that the default robots.txt will point to the auto-generated sitemap.

File: docs/guides/technical-docs-seo-guide.rst [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+
+A sitemap is a file readable by crawlers that contains a list of pages
+and other files on your site and some metadata or relationships about them
+(eg. https://docs.readthedocs.io/sitemap.xml).
+A good sitemaps provides information like how frequently a page or file is updated
+or any alternate language versions of a page.
+
+Read the Docs generates a sitemap for you that contains the last time
+your documentation was updated as well as links to active versions, subprojects, and translations your project has.
+We have a small separate guide on `sitemaps </guides/sitemaps>`_.
```
Comment (humitos) [inline docs/guides/technical-docs-seo-guide.rst]: The link does not work since it starts with `/`. I recommend using `:ref:` here.

```suggestion
We have a small separate guide on :ref:`sitemaps <guides/sitemaps>`_.
```

File: docs/guides/technical-docs-seo-guide.rst [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+        :description lang=en:
+            Your meta description here
+
+
+.. figure:: ../_static/images/guides/google-search-engine-results.png
+    :align: center
+    :figwidth: 80%
+    :target: ../_static/images/guides/google-search-engine-results.png
+
+    Google search engine results showing a customized meta description
```
Comment (humitos) [inline docs/guides/technical-docs-seo-guide.rst]: This is not a good example :smiley: 

The `description` meta tag of that page is:

```html
<meta content="How to add additional CSS stylesheets or JavaScript files to your Sphinx documentation to override your Sphinx theme or add interactivity with JavaScript." lang="en" name="description" />
```

but Google shows something different for some reason.

This blog post meta description does appear as Google's result description https://blog.readthedocs.com/custom-css-and-js-in-sphinx/

File: docs/guides/technical-docs-seo-guide.rst [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+look in search engine result pages.
+
+Typically search engines will use the first few sentences of a page if no meta description is provided.
+In Sphinx, you can customize your meta description using the following RestructuredText:
+
+.. sourcecode:: rst
+
+    .. meta::
+        :description lang=en:
+            Your meta description here
```
Comment (humitos) [inline docs/guides/technical-docs-seo-guide.rst]: I think it's better if this example reflects what the image shows.

Following my next comment about the blog post, this rst code would look like:

```rst
.. meta::
    :description lang=en:
        Customize your documentation by adding CSS stylesheets or JavaScript files to change your docs' look and feel
```

File: docs/guides/technical-docs-seo-guide.rst (hunk 2 above)
Comment (davidfischer) [reply docs/guides/technical-docs-seo-guide.rst]: This is due to this: https://github.com/readthedocs/readthedocs.org/blob/9f928343b967247d9bed8f5d7d851e38cdeef131/docs/conf.py#L59

I can test out changing it.

File: docs/guides/technical-docs-seo-guide.rst (hunk 6 above)
Comment (davidfischer) [reply docs/guides/technical-docs-seo-guide.rst]: I considered that and I think you're right at least for now. I think it's probably better to create a guide at some point though.

File: docs/guides/technical-docs-seo-guide.rst (hunk 8 above)
Comment (davidfischer) [reply docs/guides/technical-docs-seo-guide.rst]: It should be a `:doc:` link, not a ref or the broken link I did.

File: docs/guides/technical-docs-seo-guide.rst (hunk 9 above)
Comment (davidfischer) [reply docs/guides/technical-docs-seo-guide.rst]: You're right. I totally missed that. I wonder why it isn't!

Comment (davidfischer) [reply docs/guides/technical-docs-seo-guide.rst]: It looks like Google chooses whether to use the meta description based on a few factors including the length and whether the keywords in the description match the keywords on the page. Sounds like dark arts.

File: docs/guides/technical-docs-seo-guide.rst (hunk 3 above)
Comment (davidfischer) [reply docs/guides/technical-docs-seo-guide.rst]: I tried this but it doesn't come through cleanly:

Screen Shot 2019-08-19 at 8 53 39 AM

File: docs/guides/technical-docs-seo-guide.rst (hunk 6 above)
Comment (humitos) [reply docs/guides/technical-docs-seo-guide.rst]: Yes. `robots.txt` should have it's own guide, I agree.

When I wrote that FAQ I just wanted the minimal configuration required to live somewhere and I think it was a mistake to make it a FAQ, even with that small content should have been a Guide :disappointed:

File: docs/guides/technical-docs-seo-guide.rst (hunk 9 above)
Comment (humitos) [reply docs/guides/technical-docs-seo-guide.rst]: In that case, we may want to mention something "this is black magic and it may not displayed if the engine decides to not show it" :)

Otherwise, I feel that people will come back to us saying: "Hey, meta description does not work on Read the Docs" :stuck_out_tongue:

Comment (humitos) [reply docs/guides/technical-docs-seo-guide.rst]: If Google has a guide or the rules for this, we could link to that as well.

File: docs/guides/technical-docs-seo-guide.rst (hunk 3 above)
Comment (humitos) [reply docs/guides/technical-docs-seo-guide.rst]: Hrm... I think the `$` in the commands (`make`) should be omitted. Although, I'm not sure how to tell what are commands and what's output.

We can leave as you wrote it for now, though.

File: docs/guides/technical-docs-seo-guide.rst (hunk 9 above)
Comment (davidfischer) [reply docs/guides/technical-docs-seo-guide.rst]: > Otherwise, I feel that people will come back to us saying: "Hey, meta description does not work on Read the Docs" 😛

I added something to that effect yesterday 😛.

Comment (humitos) [reply docs/guides/technical-docs-seo-guide.rst]: Cool! Just saw it. :+1:

File: docs/conf.py [hunk 11]
```diff
@@ -56,7 +56,7 @@ def get_version():
 intersphinx_mapping = {
     'python': ('https://python.readthedocs.io/en/latest/', None),
     'django': ('https://django.readthedocs.io/en/1.11.x/', None),
```
Comment (ericholscher) [inline docs/conf.py]: We should probably link to the canonical docs of these other two, also.

**Comment by ericholscher** [comment]
We should also copy a lot of this content to the blog somehow :)

**Comment by davidfischer** [comment]
> We should also copy a lot of this content to the blog somehow :)

Agreed. I was planning on a short blog post linking to this. We've had a lot of blog posts recently and a couple in the pipeline so I'm not sure on timing.

**Comment by davidfischer** [comment]
> One section that I would like to add but haven't yet is a section on how good technical writing should naturally lead to better SEO.

I think a guide or maybe just a blog on technical writing (guides vs tutorials) as well as linking to resources about simpler language (good for SEO and translations) and other things is very good but probably not a fit in the SEO guide right now.

File: docs/conf.py (hunk 11 above)
Comment (davidfischer) [reply docs/conf.py]: I'll open another PR for this. This is approved so I'm going to merge it.

File: docs/guides/technical-docs-seo-guide.rst (hunk 6 above)
Comment (davidfischer) [reply docs/guides/technical-docs-seo-guide.rst]: We are trying to either create a new guide or spruce up one of the existing guides at least every month. That could be the next one!

File: docs/conf.py (hunk 11 above)
Comment (davidfischer) [reply docs/conf.py]: https://github.com/readthedocs/readthedocs.org/pull/6085
